tests/integration/workflows/test_arxiv_workflow.py

```diff
@@ -579,3 +633,96 @@ def test_harvesting_arxiv_workflow_manual_accepted(
         obj = eng.processed_objects[0]
         # It was accepted
         assert obj.status == ObjectStatus.COMPLETED
+
+
+@mock.patch(
+    'inspirehep.modules.workflows.tasks.arxiv.download_file_to_workflow',
+    side_effect=fake_download_file,
+)
+@mock.patch(
+    'inspirehep.modules.workflows.utils.download_file_to_workflow',
+    side_effect=fake_download_file,
+)
+@mock.patch(
+    'inspirehep.modules.workflows.tasks.beard.json_api_request',
+    side_effect=fake_beard_api_request,
+)
+@mock.patch(
+    'inspirehep.modules.workflows.tasks.magpie.json_api_request',
+    side_effect=fake_magpie_api_request,
+)
+@mock.patch(
+    'inspirehep.modules.authors.receivers._query_beard_api',
+    side_effect=fake_beard_api_block_request,
+)
+@mock.patch(
+    'inspirehep.modules.workflows.tasks.matching.search',
+    return_value=[],
+)
+@mock.patch(
+    'inspirehep.modules.workflows.tasks.refextract.extract_references_from_file',
+    side_effect=fake_refextract_extract_references_from_file,
+)
+def test_merge_articles(
+    mocked_refextract_extract_refs,
+    mocked_matching_search,
+    mocked_api_request_beard_block,
+    mocked_api_request_magpie,
+    mocked_api_request_beard,
+    mocked_download_utils,
+    mocked_download_arxiv,
+    small_app,
+    dummy_head,
+):
+    """Test a full harvesting workflow."""
+    def _setup_test():
+        # store head
+        # already saved by dummy_record fixture
+
+        # store root
+        record_uuid = dummy_head.id
+        original_root = read_file('fixtures', 'merger_root.json')
+
+        store_root_json(record_uuid=record_uuid, source='arXiv', json=original_root)
+        db.session.commit()
+
+    _setup_test()
+
+    update = read_file('fixtures', 'merger_update.json')
+    # this function starts the workflow
+    workflow_uuid, eng, obj = get_halted_workflow_merge(
+        app=small_app,
+        extra_config={'PRODUCTION_MODE': False},
+        record=update,
+    )
+    # here inspire-json-merger code should be executed
+    # fails because author uuid is not present in the wf
+
+    def remove_uuid_authors(record):
+        for author in record['authors']:
+            if author.get('uuid'):
+                del author['uuid']
+        return record
```

Comment:
Removes keys that depend from the database. Maybe can be written in a more compact way